### PR TITLE
GEOMESA-2467 Upgrade to geotools 20.0 and locationtech jts 1.16.0

### PR DESCRIPTION
**Pull request by jnh5y**
* [x] Enter CQs
* [x] Do more stuff.
* [x] update docs - in particular, note accumulo-client in gs must be 1.9.2, although server can be lower

Signed-off-by: Jim Hughes <jnh5y@ccri.com>

### CODE REVIEW

File: docs/user/spark/jupyter.rst [hunk 1]
```diff
@@ -136,7 +136,7 @@ The following sample notebook shows how you can use Leaflet for data visualizati
    classpath.addRepository("http://central.maven.org/maven2")
    classpath.addRepository("https://repo.locationtech.org/content/repositories/geomesa-releases")
    classpath.addRepository("file:///home/username/.m2/repository")
-   classpath.add("com.vividsolutions" % "jts" % "1.13")
+   classpath.add("org.locationtech" % "jts" % "1.13")
```
Comment (elahrvivaz) [inline docs/user/spark/jupyter.rst]: this looks like it needs to be `org.locationtech.jts`

File: docs/user/datastores/attributes.rst [hunk 2]
```diff
@@ -37,14 +37,14 @@ Boolean            java.lang.Boolean                              Yes
...
-GeometryCollection com.vividsolutions.jts.geom.GeometryCollection Yes
-Geometry           com.vividsolutions.jts.geom.Geometry           Yes
+Point              org.locationtech.jts.geom.Point              Yes
+LineString         org.locationtech.jts.geom.LineString         Yes
+Polygon            org.locationtech.jts.geom.Polygon            Yes
+MultiPoint         org.locationtech.jts.geom.MultiPoint         Yes
+MultiLineString    org.locationtech.jts.geom.MultiLineString    Yes
+MultiPolygon       org.locationtech.jts.geom.MultiPolygon       Yes
+GeometryCollection org.locationtech.jts.geom.GeometryCollection Yes
+Geometry           org.locationtech.jts.geom.Geometry           Yes
```
Comment (elahrvivaz) [inline docs/user/datastores/attributes.rst]: probably need to fix the formatting here so the table renders right

File: build/cqs.tsv [hunk 3]
```diff
@@ -101,20 +104,22 @@ it.geosolutions.jaiext.orderdither:jt-orderdither	1.0.16	compile
 it.geosolutions.jaiext.piecewise:jt-piecewise	1.0.16	compile
 it.geosolutions.jaiext.rescale:jt-rescale	1.0.16	compile
 it.geosolutions.jaiext.rlookup:jt-rlookup	1.0.16	compile
+it.geosolutions.jaiext.scale2:jt-scale2	1.1.1	compile
 it.geosolutions.jaiext.scale:jt-scale	1.0.16	compile
+it.geosolutions.jaiext.shadedrelief:jt-shadedrelief	1.1.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: these versions not matching the other jaiext jars makes me think we need to bump the version in our pom

File: build/cqs.tsv [hunk 4]
```diff
@@ -51,8 +53,8 @@ com.sun.xml.bind:jaxb-impl	2.2.11	compile
 com.thoughtworks.paranamer:paranamer	2.6	compile
 com.typesafe:config	1.3.3	compile
 com.typesafe.scala-logging:scala-logging_2.11	3.8.0	compile
-com.vividsolutions:jts-core	1.14.0	compile
-com.vividsolutions:jts-io	1.14.0	compile
+com.vividsolutions:jts	1.12	compile
+com.vividsolutions:jts	1.13	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: seems like we are still pulling in the old jts from somewhere?

**Comment by pomadchin** [comment]
Hey! What's the status of that? We'd like also to depend on it and it looks like GeoMesa is deeply integrated with GeoTools: https://github.com/locationtech/geotrellis/pull/2813

**Comment by elahrvivaz** [comment]
@pomadchin our current thinking is that we're going to release 2.1.0 in the next few weeks, and then we'll release 2.2.0 pretty shortly thereafter with this update. looks like you've already put in some CQs so that will ease our upgrade for sure.
Because of the JTS package rename, this version is likely to break a lot of user's code, or at least force them to upgrade geoserver versions.

File: docs/user/spark/jupyter.rst (hunk 1 above)
Comment (elahrvivaz) [reply docs/user/spark/jupyter.rst]: fixed

File: docs/user/datastores/attributes.rst (hunk 2 above)
Comment (elahrvivaz) [reply docs/user/datastores/attributes.rst]: fixed

File: pom.xml [hunk 5]
```diff
@@ -70,10 +70,10 @@
         <scala.parsers.version>1.0.6</scala.parsers.version>
 
         <geoserver.version>2.12.0</geoserver.version>
-        <gt.version>18.0</gt.version>
+        <gt.version>20.0</gt.version>
         <jaiext.version>1.0.16</jaiext.version>
         <imageio-ext.version>1.1.18</imageio-ext.version>
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply pom.xml]: updated to match version from geotools

File: build/cqs.tsv (hunk 3 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: fixed

File: build/cqs.tsv (hunk 4 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: fixed

File: build/cqs.tsv [hunk 6]
```diff
@@ -4,6 +4,7 @@ com.beust:jcommander	1.48	compile
 com.carrotsearch:hppc	0.7.2	compile
 com.chuusai:shapeless_2.11	2.3.2	compile
 com.clearspring.analytics:stream	2.9.2	compile
+com.conversantmedia:disruptor	1.2.13	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18153

File: build/cqs.tsv [hunk 7]
```diff
@@ -18,7 +19,7 @@ com.googlecode.concurrent-trees:concurrent-trees	2.6.0	compile
 com.googlecode.cqengine:cqengine	2.7.1	compile
 com.googlecode.efficient-java-matrix-library:core	0.26	compile
 com.google.code.gson:gson	2.8.1	compile
-com.googlecode.json-simple:json-simple	1.1	compile
+com.googlecode.json-simple:json-simple	1.1.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18154

File: build/cqs.tsv [hunk 8]
```diff
@@ -78,43 +77,45 @@ io.dropwizard.metrics:metrics-graphite	3.1.2	compile
...
+it.geosolutions.jaiext.rlookup:jt-rlookup	1.1.1	compile
+it.geosolutions.jaiext.scale2:jt-scale2	1.1.1	compile
+it.geosolutions.jaiext.scale:jt-scale	1.1.1	compile
+it.geosolutions.jaiext.shadedrelief:jt-shadedrelief	1.1.1	compile
+it.geosolutions.jaiext.stats:jt-stats	1.1.1	compile
+it.geosolutions.jaiext.translate:jt-translate	1.1.1	compile
+it.geosolutions.jaiext.utilities:jt-utilities	1.1.1	compile
+it.geosolutions.jaiext.vectorbin:jt-vectorbin	1.1.1	compile
+it.geosolutions.jaiext.warp:jt-warp	1.1.1	compile
+it.geosolutions.jaiext.zonal:jt-zonal	1.1.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18155

File: build/cqs.tsv [hunk 9]
```diff
@@ -78,43 +77,45 @@ io.dropwizard.metrics:metrics-graphite	3.1.2	compile
...
+javax.measure:unit-api	1.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18156

File: build/cqs.tsv [hunk 10]
```diff
@@ -78,43 +77,45 @@ io.dropwizard.metrics:metrics-graphite	3.1.2	compile
...
 joda-time:joda-time	2.3	compile
 log4j:log4j	1.2.17	compile
-net.java.dev.jsr-275:jsr-275	1.0-beta-2	compile
 net.jpountz.lz4:lz4	1.3.0	compile
 net.minidev:accessors-smart	1.1	compile
 net.minidev:json-smart	2.2.1	compile
-net.sf.geographiclib:GeographicLib-Java	1.44	compile
+net.sf.geographiclib:GeographicLib-Java	1.49	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18157

File: build/cqs.tsv [hunk 11]
```diff
@@ -135,11 +136,11 @@ org.apache.commons:commons-pool2	2.4.2	compile
...
 org.apache.commons:commons-vfs2	2.1	compile
 org.apache.curator:curator-client	2.4.0	compile
-org.apache.curator:curator-client	2.7.1	compile
+org.apache.curator:curator-client	4.0.1	compile
 org.apache.curator:curator-framework	2.4.0	compile
-org.apache.curator:curator-framework	2.7.1	compile
+org.apache.curator:curator-framework	4.0.1	compile
 org.apache.curator:curator-recipes	2.4.0	compile
-org.apache.curator:curator-recipes	2.7.1	compile
+org.apache.curator:curator-recipes	4.0.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18158

File: build/cqs.tsv [hunk 12]
```diff
@@ -171,41 +172,42 @@ org.eclipse.emf:org.eclipse.emf.common	2.12.0	compile
 org.eclipse.emf:org.eclipse.emf.ecore	2.12.0	compile
 org.eclipse.emf:org.eclipse.emf.ecore.xmi	2.12.0	compile
 org.eclipse.xsd:org.eclipse.xsd	2.12.0	compile
+org.ejml:ejml-core	0.34	compile
+org.ejml:ejml-ddense	0.34	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18159

Comment (jnh5y) [reply build/cqs.tsv]: @jahhulbert-ccri does this work for you?  Were you suggesting a different version of EJML?

File: build/cqs.tsv [hunk 13]
```diff
@@ -171,41 +172,42 @@ org.eclipse.emf:org.eclipse.emf.common	2.12.0	compile
...
+org.geotools.ogc:net.opengis.ows	20.0	compile
+org.geotools.ogc:net.opengis.wfs	20.0	compile
+org.geotools.ogc:org.w3.xlink	20.0	compile
+org.geotools.xsd:gt-xsd-core	20.0	compile
+org.geotools.xsd:gt-xsd-fes	20.0	compile
+org.geotools.xsd:gt-xsd-filter	20.0	compile
+org.geotools.xsd:gt-xsd-gml2	20.0	compile
+org.geotools.xsd:gt-xsd-gml3	20.0	compile
+org.geotools.xsd:gt-xsd-ows	20.0	compile
+org.geotools.xsd:gt-xsd-wfs	20.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18160

File: build/cqs.tsv [hunk 14]
```diff
@@ -234,6 +238,11 @@ org.slf4j:slf4j-api	1.7.25	compile
 org.slf4j:slf4j-log4j12	1.7.25	compile
 picocontainer:picocontainer	1.2	compile
 redis.clients:jedis	2.9.0	compile
+si.uom:si-quantity	0.7.1	compile
+si.uom:si-units-java8	0.7.1	compile
+systems.uom:systems-common-java8	0.7.2	compile
+tec.uom.lib:uom-lib-common	1.0.2	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18161

File: build/cqs.tsv [hunk 15]
```diff
@@ -234,6 +238,11 @@ org.slf4j:slf4j-api	1.7.25	compile
...
+tec.uom:uom-se	1.0.8	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18162

File: build/cqs.tsv [hunk 16]
```diff
@@ -234,6 +238,11 @@ org.slf4j:slf4j-api	1.7.25	compile
 org.slf4j:slf4j-log4j12	1.7.25	compile
 picocontainer:picocontainer	1.2	compile
 redis.clients:jedis	2.9.0	compile
+si.uom:si-quantity	0.7.1	compile
+si.uom:si-units-java8	0.7.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18163

File: build/cqs.tsv [hunk 17]
```diff
@@ -234,6 +238,11 @@ org.slf4j:slf4j-api	1.7.25	compile
...
+systems.uom:systems-common-java8	0.7.2	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18164

File: build/cqs.tsv [hunk 18]
```diff
@@ -257,8 +266,9 @@ org.apache.spark:spark-sql_2.11	2.3.1	provided
 org.apache.spark:spark-yarn_2.11	2.3.1	provided
 org.scala-lang:scala-compiler	2.11.7	provided
 org.springframework.security:spring-security-core	4.0.4.RELEASE	provided
-org.springframework.security:spring-security-web	4.0.4.RELEASE	provided
-org.springframework:spring-web	4.3.7.RELEASE	provided
+org.springframework.security:spring-security-core	4.2.7.RELEASE	provided
+org.springframework.security:spring-security-web	4.2.7.RELEASE	provided
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18165

File: build/cqs.tsv [hunk 19]
```diff
@@ -171,41 +172,42 @@ org.eclipse.emf:org.eclipse.emf.common	2.12.0	compile
...
+org.geotools.xsd:gt-xsd-fes	20.0	compile
+org.geotools.xsd:gt-xsd-filter	20.0	compile
+org.geotools.xsd:gt-xsd-gml2	20.0	compile
+org.geotools.xsd:gt-xsd-gml3	20.0	compile
+org.geotools.xsd:gt-xsd-ows	20.0	compile
+org.geotools.xsd:gt-xsd-wfs	20.0	compile
 org.hsqldb:hsqldb	2.3.0	compile
 org.jamon:jamon-runtime	2.4.1	compile
-org.jdom:jdom	1.1.3	compile
+org.jdom:jdom2	2.0.6	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18167

@jnh5y do you think it's ok to piggy back on the CQ I did?

File: build/cqs.tsv (hunk 12 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: Let's merge this as is, and I've opened a second ticket to upgrade to 0.36: https://geomesa.atlassian.net/browse/GEOMESA-2469

File: build/cqs.tsv (hunk 6 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: last one - it's still awaiting pmc approval

**Comment by elahrvivaz** [comment]
@jnh5y this is ready for merge - should probably review the diffs one last time

File: build/cqs.tsv [hunk 20]
```diff
@@ -129,17 +130,17 @@ org.apache.camel:camel-core	2.15.1	compile
 org.apache.camel:camel-scala	2.15.1	compile
 org.apache.commons:commons-compress	1.16	compile
 org.apache.commons:commons-csv	1.5	compile
-org.apache.commons:commons-lang3	3.6	compile
+org.apache.commons:commons-lang3	3.7	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18215

File: build/cqs.tsv [hunk 21]
```diff
@@ -4,21 +4,22 @@ com.beust:jcommander	1.48	compile
...
+com.conversantmedia:disruptor	1.2.13	compile
 com.drewnoakes:metadata-extractor	2.8.1	compile
 com.esotericsoftware:kryo-shaded	3.0.3	compile
 com.esotericsoftware:minlog	1.3.0	compile
-com.fasterxml.jackson.core:jackson-annotations	2.6.5	compile
-com.fasterxml.jackson.core:jackson-core	2.6.5	compile
-com.fasterxml.jackson.core:jackson-databind	2.6.5	compile
+com.fasterxml.jackson.core:jackson-annotations	2.6.7	compile
+com.fasterxml.jackson.core:jackson-core	2.6.7	compile
+com.fasterxml.jackson.core:jackson-databind	2.6.7.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18218

File: build/cqs.tsv [hunk 22]
```diff
@@ -4,21 +4,22 @@ com.beust:jcommander	1.48	compile
...
 com.chuusai:shapeless_2.11	2.3.2	compile
 com.clearspring.analytics:stream	2.9.2	compile
+com.conversantmedia:disruptor	1.2.13	compile
 com.drewnoakes:metadata-extractor	2.8.1	compile
 com.esotericsoftware:kryo-shaded	3.0.3	compile
 com.esotericsoftware:minlog	1.3.0	compile
-com.fasterxml.jackson.core:jackson-annotations	2.6.5	compile
-com.fasterxml.jackson.core:jackson-core	2.6.5	compile
-com.fasterxml.jackson.core:jackson-databind	2.6.5	compile
+com.fasterxml.jackson.core:jackson-annotations	2.6.7	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18217

File: build/cqs.tsv [hunk 23]
```diff
@@ -4,21 +4,22 @@ com.beust:jcommander	1.48	compile
...
+com.fasterxml.jackson.core:jackson-core	2.6.7	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18216

File: build/cqs.tsv [hunk 24]
```diff
@@ -4,21 +4,22 @@ com.beust:jcommander	1.48	compile
...
+com.fasterxml.jackson.core:jackson-databind	2.6.7.1	compile
 com.github.ben-manes.caffeine:caffeine	2.5.6	compile
 com.github.pureconfig:pureconfig_2.11	0.9.1	compile
 com.github.pureconfig:pureconfig-macros_2.11	0.9.1	compile
 com.github.stephenc.findbugs:findbugs-annotations	1.3.9-1	compile
-com.googlecode.concurrent-trees:concurrent-trees	2.6.0	compile
-com.googlecode.cqengine:cqengine	2.7.1	compile
+com.googlecode.concurrent-trees:concurrent-trees	2.6.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18219

File: build/cqs.tsv [hunk 25]
```diff
@@ -4,21 +4,22 @@ com.beust:jcommander	1.48	compile
...
+com.googlecode.cqengine:cqengine	3.0.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18220

File: build/cqs.tsv [hunk 26]
```diff
@@ -78,46 +77,49 @@ io.dropwizard.metrics:metrics-graphite	3.1.2	compile
...
+it.geosolutions.jaiext.translate:jt-translate	1.1.1	compile
+it.geosolutions.jaiext.utilities:jt-utilities	1.1.1	compile
+it.geosolutions.jaiext.vectorbin:jt-vectorbin	1.1.1	compile
+it.geosolutions.jaiext.warp:jt-warp	1.1.1	compile
+it.geosolutions.jaiext.zonal:jt-zonal	1.1.1	compile
+javax.measure:unit-api	1.0	compile
 joda-time:joda-time	2.3	compile
 log4j:log4j	1.2.17	compile
-net.java.dev.jsr-275:jsr-275	1.0-beta-2	compile
+net.jodah:typetools	0.5.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18221

File: build/cqs.tsv [hunk 27]
```diff
@@ -78,46 +77,49 @@ io.dropwizard.metrics:metrics-graphite	3.1.2	compile
...
 net.jpountz.lz4:lz4	1.3.0	compile
 net.minidev:accessors-smart	1.1	compile
 net.minidev:json-smart	2.2.1	compile
-net.sf.geographiclib:GeographicLib-Java	1.44	compile
+net.sf.geographiclib:GeographicLib-Java	1.49	compile
 net.sf.saxon:Saxon-HE	9.7.0-20	compile
 net.sf.trove4j:trove4j	3.0.3	compile
-org.antlr:antlr4-runtime	4.5	compile
+org.antlr:antlr4-runtime	4.7.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18222

File: build/cqs.tsv [hunk 28]
```diff
@@ -127,19 +129,19 @@ org.apache.arrow:arrow-vector	0.10.0	compile
 org.apache.avro:avro	1.7.5	compile
 org.apache.camel:camel-core	2.15.1	compile
 org.apache.camel:camel-scala	2.15.1	compile
-org.apache.commons:commons-compress	1.16	compile
+org.apache.commons:commons-compress	1.18	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18223

File: build/cqs.tsv (hunk 27 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: approved

File: build/cqs.tsv (hunk 28 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: approved

File: pom.xml [hunk 29]
```diff
@@ -1482,6 +1148,12 @@
                 <artifactId>commons-net</artifactId>
                 <version>3.3</version>
             </dependency>
+            <dependency>
+                <!-- managed from jts 1.1.1 and geotools 1.1 -->
+                <groupId>com.googlecode.json-simple</groupId>
```
Comment (jnh5y) [inline pom.xml]: Ok, am I reading this correctly that GeoTools is the project on the older version?

Comment (elahrvivaz) [reply pom.xml]: yeah, but it's just a bugfix release so hopefully will be ok

**Pull request merged**